src/dos/dos.cpp

```diff
@@ -1426,10 +1427,11 @@ static Bitu DOS_25Handler(void)
 }
 static Bitu DOS_26Handler(void) {
 	LOG(LOG_DOSMISC,LOG_NORMAL)("int 26 called: hope for the best!");
+	std::string type_fat = drive_info.at(DRIVETYPE::FAT);
 	if (reg_al >= DOS_DRIVES || !Drives[reg_al] || Drives[reg_al]->isRemovable()) {	
 		reg_ax = 0x8002;
 		SETFLAGBIT(CF,true);
-	} else if (strncmp(Drives[reg_al]->GetInfo(), "fatDrive", 8) == 0) {
+	} else if (std::string(Drives[reg_al]->GetInfo()).substr(0, type_fat.size()) == type_fat) {
```

Comment:
I see an enum has been added:

```c++
enum class DRIVETYPE : uint16_t {
	LOCAL   = 1,
	CDROM   = 2,
	FAT     = 3,
	ISO     = 4,
	VIRTUAL = 5,
};
```

And a new global variable, as map, that looks up strings:

```c++
inline std::unordered_map<DRIVETYPE, const char *> drive_info;

drive_info.insert({DRIVETYPE::LOCAL, MSG_Get("MOUNT_TYPE_LOCAL_DIRECTORY")});
drive_info.insert({DRIVETYPE::CDROM, MSG_Get("MOUNT_TYPE_CDROM")});
drive_info.insert({DRIVETYPE::FAT, MSG_Get("MOUNT_TYPE_FAT")});
drive_info.insert({DRIVETYPE::ISO, MSG_Get("MOUNT_TYPE_ISO")});
drive_info.insert({DRIVETYPE::VIRTUAL, MSG_Get("MOUNT_TYPE_VIRTUAL")});
```

Reply:
And these looked up strings are being used through the drive_info logic, just like before.

The goal of using an `enum` is avoid using string to hold type information, and replace the string comparisons with enum comparisons.

Reply:
The `else if (std::string(Drives[reg_al]->GetInfo()).substr(0, type_fat.size()) == type_fat)`

statement shouldn't be doing a string-compare. It should compare enum values.

If `->GetInfo()` previously returned a string-type, we want it to return an DriveType value. For example, `DriveType::Local`.



else if ().substr(0, type_fat.size()) == type_fat)

Reply:
So for the most part, this PR should be "de-string'ifying" the code.

The only part where we want to use the strings (the message strings), is during the logging for the user.

Reply:
@kcgen I did not use enum class that often previously, and many thanks for showing detailed examples of them. I have revised the code accordingly which now use enum comparison instead of string comparison as suggested.